projects/swimlane/ngx-charts/src/lib/common/timeline/timeline.component.ts

```diff
@@ -108,13 +110,13 @@ export class Timeline implements OnChanges {
 
     let domain = [];
     if (this.scaleType === 'time') {
-      const min = Math.min(...values);
-      const max = Math.max(...values);
+      const min = this.xScaleMin ? this.xScaleMin : Math.min(...values);
+      const max = this.xScaleMax ? this.xScaleMax : Math.max(...values);
```

Comment:
Compare with "null" to catch the value "0" with is a valid scale limit.
```suggestion
      const min = this.xScaleMin == null ? Math.min(...values) : this.xScaleMin;
      const max = this.xScaleMax == null ? Math.max(...values) : this.xScaleMax;
```

Reply:
```
const min = this.xScaleMin ?? Math.min(...values);
const max = this.xScaleMax ?? Math.max(...values);
```

Reply:
Thanks, this one is even better.